docs/src/main/sphinx/connector/iceberg.rst

```diff
@@ -127,6 +127,42 @@ supports the following features:
   :ref:`iceberg-materialized-views`
 * :ref:`sql-views-management`
 
+.. _iceberg-alter-table-execute-optimize:
+
+ALTER TABLE EXECUTE optimize
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The connector offers :ref:`ALTER TABLE EXECUTE <alter-table-execute>`
+``optimize`` command for rewriting the current content of the specified
```

Comment:
At risk of overriding your work so far, we have a similar section that documents ALTER TABLE EXECUTE for the Hive connector: https://trino.io/docs/current/connector/hive.html#alter-table-execute

For the sake of consistency, we should make these as similar as possible. Are there any differences in implementation on ALTER TABLE EXECUTE on Hive vs on Iceberg?

Reply:
> The optimize procedure is disabled by default, and can be enabled for a catalog with the `<catalog name>.non_transactional_optimize_enabled` session property.

This property is not required for the Iceberg connector.

Reply:
> Are there any differences in implementation on ALTER TABLE EXECUTE on Hive vs on Iceberg?

@findepi can you please answer this question? I am not familiar enough with the Hive implementation of `OPTIMIZE`

Reply:
> Are there any differences in implementation on ALTER TABLE EXECUTE on Hive vs on Iceberg?

They are independent. Hive's is a dangerous, non-transactional operation that can, in case of failure, lead to corrupted table state.

Reply:
> > Are there any differences in implementation on ALTER TABLE EXECUTE on Hive vs on Iceberg?
> 
> They are independent. Hive's is a dangerous, non-transactional operation that can, in case of failure, lead to corrupted table state.

Would the content in https://trino.io/docs/current/connector/hive.html#alter-table-execute make sense here then, just without the warning note about it being a non-transactional operation?

Reply:
@jhlodin mostly, yes, except  for minor wording https://github.com/trinodb/trino/pull/10790#discussion_r793529499

Reply:
@findinpath Can you update the contents of this commit to reuse the Hive documentation? Minus the note as mentioned, and with whatever other feedback there is.

Reply:
I have aligned both Hive and Iceberg documentation regarding `optimize` command to follow the same structure.
I've taken the liberty of adding along the way a few more usage examples for the command in the Hive documentation.

Reply:
Looks great, thanks!